<Type Name="AggregateException" FullName="System.AggregateException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="a408cd8c65b6a1872bbc7ecc28fd7106d4fe9897" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="zh-CN" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class AggregateException : Exception" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit AggregateException extends System.Exception" />
  <TypeSignature Language="DocId" Value="T:System.AggregateException" />
  <TypeSignature Language="VB.NET" Value="Public Class AggregateException&#xA;Inherits Exception" />
  <TypeSignature Language="C++ CLI" Value="public ref class AggregateException : Exception" />
  <AssemblyInfo>
    <AssemblyName>System.Threading.Tasks</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Diagnostics.DebuggerDisplay("Count = {InnerExceptionCount}")</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="3d01a-101">表示在应用程序执行过程中发生的一个或多个错误。</span>
      <span class="sxs-lookup">
        <span data-stu-id="3d01a-101">Represents one or more errors that occur during application execution.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="3d01a-102"><xref:System.AggregateException> 用于将多个故障整合到单个可引发的异常对象。</span><span class="sxs-lookup"><span data-stu-id="3d01a-102"><xref:System.AggregateException> is used to consolidate multiple failures into a single, throwable exception object.</span></span> <span data-ttu-id="3d01a-103">在广泛使用[任务并行库 (TPL)](~/docs/standard/parallel-programming/task-parallel-library-tpl.md)和[并行 LINQ (PLINQ)](~/docs/standard/parallel-programming/parallel-linq-plinq.md)。</span><span class="sxs-lookup"><span data-stu-id="3d01a-103">It is used extensively in the [Task Parallel Library (TPL)](~/docs/standard/parallel-programming/task-parallel-library-tpl.md) and [Parallel LINQ (PLINQ)](~/docs/standard/parallel-programming/parallel-linq-plinq.md).</span></span> <span data-ttu-id="3d01a-104">有关其他信息，请参阅[聚合异常](http://go.microsoft.com/fwlink/p/?LinkId=259617).NET 之所以重要博客中的条目。</span><span class="sxs-lookup"><span data-stu-id="3d01a-104">For additional information, see the [Aggregating Exceptions](http://go.microsoft.com/fwlink/p/?LinkId=259617) entry in the .NET Matters blog.</span></span> <span data-ttu-id="3d01a-105">有关示例，请参阅[NIB： 如何： 处理由任务引发异常](http://msdn.microsoft.com/library/d6c47ec8-9de9-4880-beb3-ff19ae51565d)和[如何： 处理 PLINQ 查询中的异常](~/docs/standard/parallel-programming/how-to-handle-exceptions-in-a-plinq-query.md)。</span><span class="sxs-lookup"><span data-stu-id="3d01a-105">For an example, see [NIB: How to: Handle Exceptions Thrown by Tasks](http://msdn.microsoft.com/library/d6c47ec8-9de9-4880-beb3-ff19ae51565d) and [How to: Handle Exceptions in a PLINQ Query](~/docs/standard/parallel-programming/how-to-handle-exceptions-in-a-plinq-query.md).</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="3d01a-106">下面的示例捕获<xref:System.AggregateException>异常和调用<xref:System.AggregateException.Handle%2A>方法以处理它包含每个异常。</span><span class="sxs-lookup"><span data-stu-id="3d01a-106">The following example catches the <xref:System.AggregateException> exception and calls the <xref:System.AggregateException.Handle%2A> method to handle each exception it contains.</span></span> <span data-ttu-id="3d01a-107">编译并运行该示例与第一个`task1`变量应导致<xref:System.AggregateException>对象，其中包含<xref:System.UnauthorizedAccessException>异常。</span><span class="sxs-lookup"><span data-stu-id="3d01a-107">Compiling and running the example with the first `task1` variable should result in an <xref:System.AggregateException> object that contains an <xref:System.UnauthorizedAccessException> exception.</span></span> <span data-ttu-id="3d01a-108">注释掉该行，取消注释第二个`task1`变量和编译和运行示例生成<xref:System.AggregateException>对象，其中包含<xref:System.IndexOutOfRangeException>异常。</span><span class="sxs-lookup"><span data-stu-id="3d01a-108">Commenting out that line, uncommenting the second `task1` variable, and compiling and running the example produces an <xref:System.AggregateException> object that contains an <xref:System.IndexOutOfRangeException> exception.</span></span>  
  
 [!code-csharp[System.AggregateException.Class#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.aggregateexception.class/cs/exception1.cs#1)]
 [!code-vb[System.AggregateException.Class#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.aggregateexception.class/vb/exception1.vb#1)]  
  
 ]]></format>
    </remarks>
    <threadsafe>
      <span data-ttu-id="3d01a-109">所有公共和受保护成员<see cref="T:System.AggregateException" />是线程安全的可从多个线程同时使用。</span>
      <span class="sxs-lookup">
        <span data-stu-id="3d01a-109">All public and protected members of <see cref="T:System.AggregateException" /> are thread-safe and may be used concurrently from multiple threads.</span>
      </span>
    </threadsafe>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="3d01a-110">初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-110">Initializes a new instance of the <see cref="T:System.AggregateException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AggregateException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AggregateException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="3d01a-111">使用由系统提供的用于描述错误的消息初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-111">Initializes a new instance of the <see cref="T:System.AggregateException" /> class with a system-supplied message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="3d01a-112">此构造函数初始化<xref:System.Exception.Message%2A>为系统提供的消息，如描述错误，将新实例的属性"System.AggregateException： 发生了一个或多个错误。"</span><span class="sxs-lookup"><span data-stu-id="3d01a-112">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "System.AggregateException: One or more errors occurred."</span></span> <span data-ttu-id="3d01a-113">此消息将当前系统区域性考虑在内。</span><span class="sxs-lookup"><span data-stu-id="3d01a-113">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="3d01a-114">下表显示 <xref:System.AggregateException> 的实例的初始属性值。</span><span class="sxs-lookup"><span data-stu-id="3d01a-114">The following table shows the initial property values for an instance of <xref:System.AggregateException>.</span></span>  
  
|<span data-ttu-id="3d01a-115">Property</span><span class="sxs-lookup"><span data-stu-id="3d01a-115">Property</span></span>|<span data-ttu-id="3d01a-116">“值”</span><span class="sxs-lookup"><span data-stu-id="3d01a-116">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="3d01a-117">`null`。</span><span class="sxs-lookup"><span data-stu-id="3d01a-117">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="3d01a-118">本地化的错误消息字符串。</span><span class="sxs-lookup"><span data-stu-id="3d01a-118">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AggregateException (System.Collections.Generic.IEnumerable&lt;Exception&gt; innerExceptions);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(class System.Collections.Generic.IEnumerable`1&lt;class System.Exception&gt; innerExceptions) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.#ctor(System.Collections.Generic.IEnumerable{System.Exception})" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (innerExceptions As IEnumerable(Of Exception))" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AggregateException(System::Collections::Generic::IEnumerable&lt;Exception ^&gt; ^ innerExceptions);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="innerExceptions" Type="System.Collections.Generic.IEnumerable&lt;System.Exception&gt;" />
      </Parameters>
      <Docs>
        <param name="innerExceptions">
          <span data-ttu-id="3d01a-119">导致当前异常的异常。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-119">The exceptions that are the cause of the current exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="3d01a-120">用对作为此异常原因的内部异常的引用初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-120">Initializes a new instance of the <see cref="T:System.AggregateException" /> class with references to the inner exceptions that are the cause of this exception.</span>
          </span>
        </summary>
        <remarks>To be added.</remarks>
        <exception cref="T:System.ArgumentNullException">
          <span data-ttu-id="3d01a-121">
            <paramref name="innerExceptions" /> 参数为 null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-121">The <paramref name="innerExceptions" /> argument is null.</span>
          </span>
        </exception>
        <exception cref="T:System.ArgumentException">
          <span data-ttu-id="3d01a-122">
            <paramref name="innerExceptions" /> 的元素为 Null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-122">An element of <paramref name="innerExceptions" /> is null.</span>
          </span>
        </exception>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AggregateException (params Exception[] innerExceptions);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(class System.Exception[] innerExceptions) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.#ctor(System.Exception[])" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (ParamArray innerExceptions As Exception())" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AggregateException(... cli::array &lt;Exception ^&gt; ^ innerExceptions);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="innerExceptions" Type="System.Exception[]">
          <Attributes>
            <Attribute>
              <AttributeName>System.ParamArray</AttributeName>
            </Attribute>
          </Attributes>
        </Parameter>
      </Parameters>
      <Docs>
        <param name="innerExceptions">
          <span data-ttu-id="3d01a-123">导致当前异常的异常。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-123">The exceptions that are the cause of the current exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="3d01a-124">用对作为此异常原因的内部异常的引用初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-124">Initializes a new instance of the <see cref="T:System.AggregateException" /> class with references to the inner exceptions that are the cause of this exception.</span>
          </span>
        </summary>
        <remarks>To be added.</remarks>
        <exception cref="T:System.ArgumentNullException">
          <span data-ttu-id="3d01a-125">
            <paramref name="innerExceptions" /> 参数为 null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-125">The <paramref name="innerExceptions" /> argument is null.</span>
          </span>
        </exception>
        <exception cref="T:System.ArgumentException">
          <span data-ttu-id="3d01a-126">
            <paramref name="innerExceptions" /> 的元素为 Null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-126">An element of <paramref name="innerExceptions" /> is null.</span>
          </span>
        </exception>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AggregateException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AggregateException(System::String ^ message);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="3d01a-127">描述该异常的消息。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-127">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="3d01a-128">此构造函数的调用方需要确保此字符串已针对当前系统区域性进行了本地化。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-128">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="3d01a-129">使用指定的描述错误的消息初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-129">Initializes a new instance of the <see cref="T:System.AggregateException" /> class with a specified message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="3d01a-130">下表显示 <xref:System.AggregateException> 的实例的初始属性值。</span><span class="sxs-lookup"><span data-stu-id="3d01a-130">The following table shows the initial property values for an instance of <xref:System.AggregateException>.</span></span>  
  
|<span data-ttu-id="3d01a-131">Property</span><span class="sxs-lookup"><span data-stu-id="3d01a-131">Property</span></span>|<span data-ttu-id="3d01a-132">“值”</span><span class="sxs-lookup"><span data-stu-id="3d01a-132">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="3d01a-133">`null`。</span><span class="sxs-lookup"><span data-stu-id="3d01a-133">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="3d01a-134">`message` 中指定的错误消息字符串。</span><span class="sxs-lookup"><span data-stu-id="3d01a-134">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected AggregateException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; AggregateException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute>
          <AttributeName>System.Security.SecurityCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="3d01a-135">承载序列化对象数据的对象。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-135">The object that holds the serialized object data.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="3d01a-136">关于来源和目标的上下文信息</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-136">The contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="3d01a-137">用序列化数据初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-137">Initializes a new instance of the <see cref="T:System.AggregateException" /> class with serialized data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="3d01a-138">此构造函数在反序列化过程中调用，以便重建通过流传输的异常对象。</span><span class="sxs-lookup"><span data-stu-id="3d01a-138">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="3d01a-139">有关详细信息，请参阅[XML 和 SOAP 序列化](~/docs/standard/serialization/xml-and-soap-serialization.md)。</span><span class="sxs-lookup"><span data-stu-id="3d01a-139">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <exception cref="T:System.ArgumentNullException">
          <span data-ttu-id="3d01a-140">
            <paramref name="info" /> 参数为 null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-140">The <paramref name="info" /> argument is null.</span>
          </span>
        </exception>
        <exception cref="T:System.Runtime.Serialization.SerializationException">
          <span data-ttu-id="3d01a-141">无法正确反序列化异常。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-141">The exception could not be deserialized correctly.</span>
          </span>
        </exception>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AggregateException (string message, System.Collections.Generic.IEnumerable&lt;Exception&gt; innerExceptions);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Collections.Generic.IEnumerable`1&lt;class System.Exception&gt; innerExceptions) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.#ctor(System.String,System.Collections.Generic.IEnumerable{System.Exception})" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerExceptions As IEnumerable(Of Exception))" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AggregateException(System::String ^ message, System::Collections::Generic::IEnumerable&lt;Exception ^&gt; ^ innerExceptions);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerExceptions" Type="System.Collections.Generic.IEnumerable&lt;System.Exception&gt;" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="3d01a-142">解释异常原因的错误消息。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-142">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerExceptions">
          <span data-ttu-id="3d01a-143">导致当前异常的异常。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-143">The exceptions that are the cause of the current exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="3d01a-144">使用指定错误信息和对作为此异常原因的内部异常的引用来初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-144">Initializes a new instance of the <see cref="T:System.AggregateException" /> class with a specified error message and references to the inner exceptions that are the cause of this exception.</span>
          </span>
        </summary>
        <remarks>To be added.</remarks>
        <exception cref="T:System.ArgumentNullException">
          <span data-ttu-id="3d01a-145">
            <paramref name="innerExceptions" /> 参数为 null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-145">The <paramref name="innerExceptions" /> argument is null.</span>
          </span>
        </exception>
        <exception cref="T:System.ArgumentException">
          <span data-ttu-id="3d01a-146">
            <paramref name="innerExceptions" /> 的元素为 Null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-146">An element of <paramref name="innerExceptions" /> is null.</span>
          </span>
        </exception>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AggregateException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AggregateException(System::String ^ message, Exception ^ innerException);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="3d01a-147">描述该异常的消息。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-147">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="3d01a-148">此构造函数的调用方需要确保此字符串已针对当前系统区域性进行了本地化。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-148">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="3d01a-149">导致当前异常的异常。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-149">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="3d01a-150">如果 <c>innerException</c> 参数不为 <see langword="null" />，则当前异常会在处理内部异常的 <see langword="catch" /> 块中引发。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-150">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="3d01a-151">使用指定的错误消息和对作为此异常原因的内部异常的引用来初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-151">Initializes a new instance of the <see cref="T:System.AggregateException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="3d01a-152">因前一个异常而直接引发的异常应在 <xref:System.Exception.InnerException%2A> 属性中包含对前一个异常的引用。</span><span class="sxs-lookup"><span data-stu-id="3d01a-152">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="3d01a-153"><xref:System.Exception.InnerException%2A> 属性返回的值与传递到构造函数中的值相同；或者，如果 `null` 属性没有向构造函数提供内部异常值，则为 <xref:System.Exception.InnerException%2A>。</span><span class="sxs-lookup"><span data-stu-id="3d01a-153">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="3d01a-154">下表显示 <xref:System.AggregateException> 的实例的初始属性值。</span><span class="sxs-lookup"><span data-stu-id="3d01a-154">The following table shows the initial property values for an instance of <xref:System.AggregateException>.</span></span>  
  
|<span data-ttu-id="3d01a-155">Property</span><span class="sxs-lookup"><span data-stu-id="3d01a-155">Property</span></span>|<span data-ttu-id="3d01a-156">“值”</span><span class="sxs-lookup"><span data-stu-id="3d01a-156">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="3d01a-157">`null`。</span><span class="sxs-lookup"><span data-stu-id="3d01a-157">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="3d01a-158">`message` 中指定的错误消息字符串。</span><span class="sxs-lookup"><span data-stu-id="3d01a-158">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <exception cref="T:System.ArgumentNullException">
          <span data-ttu-id="3d01a-159">
            <paramref name="innerException" /> 参数为 null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-159">The <paramref name="innerException" /> argument is null.</span>
          </span>
        </exception>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AggregateException (string message, params Exception[] innerExceptions);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception[] innerExceptions) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.#ctor(System.String,System.Exception[])" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, ParamArray innerExceptions As Exception())" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AggregateException(System::String ^ message, ... cli::array &lt;Exception ^&gt; ^ innerExceptions);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerExceptions" Type="System.Exception[]">
          <Attributes>
            <Attribute>
              <AttributeName>System.ParamArray</AttributeName>
            </Attribute>
          </Attributes>
        </Parameter>
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="3d01a-160">解释异常原因的错误消息。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-160">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerExceptions">
          <span data-ttu-id="3d01a-161">导致当前异常的异常。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-161">The exceptions that are the cause of the current exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="3d01a-162">使用指定错误信息和对作为此异常原因的内部异常的引用来初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-162">Initializes a new instance of the <see cref="T:System.AggregateException" /> class with a specified error message and references to the inner exceptions that are the cause of this exception.</span>
          </span>
        </summary>
        <remarks>To be added.</remarks>
        <exception cref="T:System.ArgumentNullException">
          <span data-ttu-id="3d01a-163">
            <paramref name="innerExceptions" /> 参数为 null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-163">The <paramref name="innerExceptions" /> argument is null.</span>
          </span>
        </exception>
        <exception cref="T:System.ArgumentException">
          <span data-ttu-id="3d01a-164">
            <paramref name="innerExceptions" /> 的元素为 Null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-164">An element of <paramref name="innerExceptions" /> is null.</span>
          </span>
        </exception>
      </Docs>
    </Member>
    <Member MemberName="Flatten">
      <MemberSignature Language="C#" Value="public AggregateException Flatten ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig instance class System.AggregateException Flatten() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.Flatten" />
      <MemberSignature Language="VB.NET" Value="Public Function Flatten () As AggregateException" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AggregateException ^ Flatten();" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.AggregateException</ReturnType>
      </ReturnValue>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="3d01a-165">将 <see cref="T:System.AggregateException" /> 实例平展入单个新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-165">Flattens an <see cref="T:System.AggregateException" /> instances into a single, new instance.</span>
          </span>
        </summary>
        <returns>
          <span data-ttu-id="3d01a-166">一个新的平展 <see cref="T:System.AggregateException" />。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-166">A new, flattened <see cref="T:System.AggregateException" />.</span>
          </span>
        </returns>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="3d01a-167">如果某个任务具有引发异常的附加的子任务，该异常包装在<xref:System.AggregateException>异常传播到父任务，将该异常包装在自己之前<xref:System.AggregateException>异常之前将其传播回调用线程。</span><span class="sxs-lookup"><span data-stu-id="3d01a-167">If a task has an attached child task that throws an exception, that exception is wrapped in an <xref:System.AggregateException> exception before it is propagated to the parent task, which wraps that exception in its own <xref:System.AggregateException> exception before it propagates it back to the calling thread.</span></span> <span data-ttu-id="3d01a-168">在这种情况下，<xref:System.AggregateException.InnerExceptions%2A>属性<xref:System.AggregateException>由捕获异常<xref:System.Threading.Tasks.Task.Wait%2A?displayProperty=nameWithType>， <xref:System.Threading.Tasks.Task.Wait*>，<xref:System.Threading.Tasks.Task.WaitAny%2A?displayProperty=nameWithType>或<xref:System.Threading.Tasks.Task.WaitAll%2A?displayProperty=nameWithType>方法包含一个或多个<xref:System.AggregateException>实例，不是原始的异常，导致错误。</span><span class="sxs-lookup"><span data-stu-id="3d01a-168">In such cases, the <xref:System.AggregateException.InnerExceptions%2A> property of the <xref:System.AggregateException> exception that is caught by the <xref:System.Threading.Tasks.Task.Wait%2A?displayProperty=nameWithType>, <xref:System.Threading.Tasks.Task.Wait*>, <xref:System.Threading.Tasks.Task.WaitAny%2A?displayProperty=nameWithType> or <xref:System.Threading.Tasks.Task.WaitAll%2A?displayProperty=nameWithType> method contains one or more <xref:System.AggregateException> instances, not the original exceptions that caused the fault.</span></span> <span data-ttu-id="3d01a-169">为了避免必须循环访问嵌套<xref:System.AggregateException>例外情况之外，你可以使用<xref:System.AggregateException.Flatten%2A>方法删除所有嵌套<xref:System.AggregateException>异常，以便<xref:System.AggregateException.InnerExceptions%2A>属性返回的<xref:System.AggregateException>对象包含原始异常。</span><span class="sxs-lookup"><span data-stu-id="3d01a-169">To avoid having to iterate over nested <xref:System.AggregateException> exceptions, you can use the <xref:System.AggregateException.Flatten%2A> method to remove all the nested <xref:System.AggregateException> exceptions, so that the <xref:System.AggregateException.InnerExceptions%2A> property of the returned <xref:System.AggregateException> object contains the original exceptions.</span></span>  
  
 <span data-ttu-id="3d01a-170">此方法以递归方式将平展的所有实例<xref:System.AggregateException>是当前的内部异常的异常<xref:System.AggregateException>实例。</span><span class="sxs-lookup"><span data-stu-id="3d01a-170">This method recursively flattens all instances of  <xref:System.AggregateException> exceptions that are inner exceptions of the current <xref:System.AggregateException> instance .</span></span> <span data-ttu-id="3d01a-171">在新的内部异常返回<xref:System.AggregateException>异常树中的所有内部异常的联合位于当前是否<xref:System.AggregateException>实例。</span><span class="sxs-lookup"><span data-stu-id="3d01a-171">The inner exceptions returned in the new <xref:System.AggregateException> are the union of all the inner exceptions from exception tree rooted at the current <xref:System.AggregateException> instance.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="3d01a-172">在下面的示例中，嵌套 <xref:System.AggregateException> 实例已经平展，并且仅在一个循环中处理。</span><span class="sxs-lookup"><span data-stu-id="3d01a-172">In the following example, nested <xref:System.AggregateException> instances are flattened and handled in just one loop.</span></span>  
  
 [!code-csharp[TPL_Exceptions#22](~/samples/snippets/csharp/VS_Snippets_Misc/tpl_exceptions/cs/flatten2.cs#22)]
 [!code-vb[TPL_Exceptions#22](~/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_exceptions/vb/flatten2.vb#22)]  
  
 <span data-ttu-id="3d01a-173">还可以使用 <xref:System.AggregateException.Flatten%2A?displayProperty=nameWithType> 方法，通过多个任务在一个 <xref:System.AggregateException> 实例中抛出的多个 <xref:System.AggregateException> 实例重新抛出内部异常，如下面的示例所示。</span><span class="sxs-lookup"><span data-stu-id="3d01a-173">You can also use the <xref:System.AggregateException.Flatten%2A?displayProperty=nameWithType> method to rethrow the inner exceptions from multiple <xref:System.AggregateException> instances thrown by multiple tasks in a single <xref:System.AggregateException> instance, as the following example shows.</span></span>  
  
 [!code-csharp[TPL_Exceptions#13](~/samples/snippets/csharp/VS_Snippets_Misc/tpl_exceptions/cs/taskexceptions2.cs#13)]
 [!code-vb[TPL_Exceptions#13](~/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_exceptions/vb/taskexceptions2.vb#13)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="GetBaseException">
      <MemberSignature Language="C#" Value="public override Exception GetBaseException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig virtual instance class System.Exception GetBaseException() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.GetBaseException" />
      <MemberSignature Language="VB.NET" Value="Public Overrides Function GetBaseException () As Exception" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; override Exception ^ GetBaseException();" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Exception</ReturnType>
      </ReturnValue>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="3d01a-174">返回 <see cref="T:System.AggregateException" />，它是此异常的根本原因。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-174">Returns the <see cref="T:System.AggregateException" /> that is the root cause of this exception.</span>
          </span>
        </summary>
        <returns>
          <span data-ttu-id="3d01a-175">返回 <see cref="T:System.AggregateException" />，它是此异常的根本原因。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-175">Returns the <see cref="T:System.AggregateException" /> that is the root cause of this exception.</span>
          </span>
        </returns>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName="GetObjectData">
      <MemberSignature Language="C#" Value="public override void GetObjectData (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig virtual instance void GetObjectData(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Public Overrides Sub GetObjectData (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; override void GetObjectData(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute>
          <AttributeName>System.Security.SecurityCritical</AttributeName>
        </Attribute>
      </Attributes>
      <ReturnValue>
        <ReturnType>System.Void</ReturnType>
      </ReturnValue>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="3d01a-176">承载序列化对象数据的对象。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-176">The object that holds the serialized object data.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="3d01a-177">关于来源和目标的上下文信息</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-177">The contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="3d01a-178">用序列化数据初始化 <see cref="T:System.AggregateException" /> 类的新实例。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-178">Initializes a new instance of the <see cref="T:System.AggregateException" /> class with serialized data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="3d01a-179">此构造函数在反序列化过程中调用，以便重建通过流传输的异常对象。</span><span class="sxs-lookup"><span data-stu-id="3d01a-179">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="3d01a-180">有关详细信息，请参阅[XML 和 SOAP 序列化](~/docs/standard/serialization/xml-and-soap-serialization.md)。</span><span class="sxs-lookup"><span data-stu-id="3d01a-180">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <exception cref="T:System.ArgumentNullException">
          <span data-ttu-id="3d01a-181">
            <paramref name="info" /> 参数为 null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-181">The <paramref name="info" /> argument is null.</span>
          </span>
        </exception>
      </Docs>
    </Member>
    <Member MemberName="Handle">
      <MemberSignature Language="C#" Value="public void Handle (Func&lt;Exception,bool&gt; predicate);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig instance void Handle(class System.Func`2&lt;class System.Exception, bool&gt; predicate) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.Handle(System.Func{System.Exception,System.Boolean})" />
      <MemberSignature Language="VB.NET" Value="Public Sub Handle (predicate As Func(Of Exception, Boolean))" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; void Handle(Func&lt;Exception ^, bool&gt; ^ predicate);" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Void</ReturnType>
      </ReturnValue>
      <Parameters>
        <Parameter Name="predicate" Type="System.Func&lt;System.Exception,System.Boolean&gt;" />
      </Parameters>
      <Docs>
        <param name="predicate">
          <span data-ttu-id="3d01a-182">每个异常要执行的谓词。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-182">The predicate to execute for each exception.</span>
          </span>
          <span data-ttu-id="3d01a-183">该谓词作为要处理的 <see cref="T:System.Exception" /> 的参数并返回一个布尔值以指示是否处理了该异常。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-183">The predicate accepts as an argument the <see cref="T:System.Exception" /> to be processed and returns a Boolean to indicate whether the exception was handled.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="3d01a-184">在每个由此 <see cref="T:System.AggregateException" /> 包含的 <see cref="T:System.Exception" /> 上调用处理程序。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-184">Invokes a handler on each <see cref="T:System.Exception" /> contained by this <see cref="T:System.AggregateException" />.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="3d01a-185">每次调用`predicate`，则返回 true 或 false 以指示是否<xref:System.Exception>已处理。</span><span class="sxs-lookup"><span data-stu-id="3d01a-185">Each invocation of the `predicate` returns true or false to indicate whether the <xref:System.Exception> was handled.</span></span> <span data-ttu-id="3d01a-186">在所有调用后任何异常未处理，如果所有未经处理的异常将会放入新<xref:System.AggregateException>将引发。</span><span class="sxs-lookup"><span data-stu-id="3d01a-186">After all invocations, if any exceptions went unhandled, all unhandled exceptions will be put into a new <xref:System.AggregateException> which will be thrown.</span></span> <span data-ttu-id="3d01a-187">否则为<xref:System.AggregateException.Handle%2A>方法只返回。</span><span class="sxs-lookup"><span data-stu-id="3d01a-187">Otherwise, the <xref:System.AggregateException.Handle%2A> method simply returns.</span></span> <span data-ttu-id="3d01a-188">如果的任何调用`predicate`引发异常，它将暂停的更多的异常处理，并立即传播引发的异常-是。</span><span class="sxs-lookup"><span data-stu-id="3d01a-188">If any invocations of the `predicate` throws an exception, it will halt the processing of any more exceptions and immediately propagate the thrown exception as-is.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="3d01a-189">通常，异常处理程序捕获<xref:System.AggregateException>异常使用`foreach`循环 （在 C# 中) 或`For Each`循环 （在 Visual Basic 中) 来处理每个异常中的其<xref:System.AggregateException.InnerExceptions%2A>集合。</span><span class="sxs-lookup"><span data-stu-id="3d01a-189">Ordinarily, an exception handler that catches an <xref:System.AggregateException> exception uses a `foreach` loop (in C#) or `For Each` loop (in Visual Basic) to handle each exception in its <xref:System.AggregateException.InnerExceptions%2A> collection.</span></span> <span data-ttu-id="3d01a-190">相反，下面的示例使用<xref:System.AggregateException.Handle%2A>方法以处理每个异常，并且不是仅重新引发的异常`CustomException`实例。</span><span class="sxs-lookup"><span data-stu-id="3d01a-190">Instead, the following example uses the <xref:System.AggregateException.Handle%2A> method to handle each exception, and only re-throws exceptions that are not `CustomException` instances.</span></span>  
  
 [!code-csharp[TPL_Exceptions#16](~/samples/snippets/csharp/VS_Snippets_Misc/tpl_exceptions/cs/handlemethod2.cs#16)]
 [!code-vb[TPL_Exceptions#16](~/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_exceptions/vb/handlemethod2.vb#16)]  
  
 <span data-ttu-id="3d01a-191">以下是使用的更完整示例<xref:System.AggregateException.Handle%2A>方法以提供对特殊处理<xref:System.UnauthorizedAccessException>枚举文件时。</span><span class="sxs-lookup"><span data-stu-id="3d01a-191">The following is a more complete example that uses the <xref:System.AggregateException.Handle%2A> method to provide special handling for an <xref:System.UnauthorizedAccessException> when enumerating files.</span></span>  
  
 [!code-csharp[TPL_Exceptions#12](~/samples/snippets/csharp/VS_Snippets_Misc/tpl_exceptions/cs/taskexceptions.cs#12)]
 [!code-vb[TPL_Exceptions#12](~/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_exceptions/vb/taskexceptions.vb#12)]  
  
 ]]></format>
        </remarks>
        <exception cref="T:System.ArgumentNullException">
          <span data-ttu-id="3d01a-192">
            <paramref name="predicate" /> 参数为 null。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-192">The <paramref name="predicate" /> argument is null.</span>
          </span>
        </exception>
        <exception cref="T:System.AggregateException">
          <span data-ttu-id="3d01a-193">未处理此 <see cref="T:System.AggregateException" /> 包含的异常。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-193">An exception contained by this <see cref="T:System.AggregateException" /> was not handled.</span>
          </span>
        </exception>
      </Docs>
    </Member>
    <Member MemberName="InnerExceptions">
      <MemberSignature Language="C#" Value="public System.Collections.ObjectModel.ReadOnlyCollection&lt;Exception&gt; InnerExceptions { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance class System.Collections.ObjectModel.ReadOnlyCollection`1&lt;class System.Exception&gt; InnerExceptions" />
      <MemberSignature Language="DocId" Value="P:System.AggregateException.InnerExceptions" />
      <MemberSignature Language="VB.NET" Value="Public ReadOnly Property InnerExceptions As ReadOnlyCollection(Of Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; property System::Collections::ObjectModel::ReadOnlyCollection&lt;Exception ^&gt; ^ InnerExceptions { System::Collections::ObjectModel::ReadOnlyCollection&lt;Exception ^&gt; ^ get(); };" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Collections.ObjectModel.ReadOnlyCollection&lt;System.Exception&gt;</ReturnType>
      </ReturnValue>
      <Docs>
        <summary>
          <span data-ttu-id="3d01a-194">获取导致当前异常的 <see cref="T:System.Exception" /> 实例的只读集合。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-194">Gets a read-only collection of the <see cref="T:System.Exception" /> instances that caused the current exception.</span>
          </span>
        </summary>
        <value>
          <span data-ttu-id="3d01a-195">返回导致当前异常的 <see cref="T:System.Exception" /> 实例的只读集合。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-195">Returns a read-only collection of the <see cref="T:System.Exception" /> instances that caused the current exception.</span>
          </span>
        </value>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName="Message">
      <MemberSignature Language="C#" Value="public override string Message { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance string Message" />
      <MemberSignature Language="DocId" Value="P:System.AggregateException.Message" />
      <MemberSignature Language="VB.NET" Value="Public Overrides ReadOnly Property Message As String" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; virtual property System::String ^ Message { System::String ^ get(); };" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.String</ReturnType>
      </ReturnValue>
      <Docs>
        <summary>To be added.</summary>
        <value>To be added.</value>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName="ToString">
      <MemberSignature Language="C#" Value="public override string ToString ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig virtual instance string ToString() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AggregateException.ToString" />
      <MemberSignature Language="VB.NET" Value="Public Overrides Function ToString () As String" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; override System::String ^ ToString();" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Tasks</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.String</ReturnType>
      </ReturnValue>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="3d01a-196">创建并返回当前 <see cref="T:System.AggregateException" /> 的字符串表示形式。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-196">Creates and returns a string representation of the current <see cref="T:System.AggregateException" />.</span>
          </span>
        </summary>
        <returns>
          <span data-ttu-id="3d01a-197">当前异常的字符串表示形式。</span>
          <span class="sxs-lookup">
            <span data-stu-id="3d01a-197">A string representation of the current exception.</span>
          </span>
        </returns>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
  </Members>
</Type>